assets/aml-benchmark/components/src/aml_benchmark/prompt_crafter/package/prompt_factory.py

```diff
@@ -199,6 +200,13 @@ def process_row(self, row: Dict, index: int) -> Dict:
                 raise BenchmarkValidationException._with_error(
                     AzureMLError.create(BenchmarkValidationError, error_details=mssg))
 
+        if self.additional_columns:
+            for k in self.additional_columns.split(","):
+                if k in row:
+                    output_data[k] = row[k]
+                else:
+                    logger.warning(f"Column name {k} not found in data at row {index}")
```

Comment:
This should be a user error instead of a warning.

Reply:
This is actually by design. We don't want to fail the entire run if user just wrongly specified additional columns, as it would only impact how gpt star metrics are calculated.

What do you think, @EricWrightAtWork , @jialiu103 ?

Reply:
Here, current implementation can lead to failures in components down the line while current component passes with a warning buried in the logs. This will lead us/users looking in wrong component for the issue when the component at fault silently ignored it.

An example- Let's say I misspell a column by mistake in prompt-crafter. The whole pipeline will finish without error and compute-metrics will likely fail (or might even silently pass not calculating some metric). User will try to debug prompt-crafter when they should be looking at prompt-crafter.

In my opinion, I would suggest showing a user error so that it fails quickly giving user opportunity to correct it.